### PR TITLE
Add tasks to create images with tests

### PR DESCRIPTION
**Pull request by uarlouski**

### CODE REVIEW

File: 2.0/gradle/vividus-test-docker.gradle [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    }
+    dependencies {
+        classpath(group: 'com.bmuschko', name: 'gradle-docker-plugin', version: '9.3.4')
+    }
+}
+
+apply plugin: com.bmuschko.gradle.docker.DockerRemoteApiPlugin
+
+tasks.register('copyResources', Copy) {
+    into layout.buildDirectory.dir('docker/resources')
```
Comment (valfirst) [inline 2.0/gradle/vividus-test-docker.gradle:29]: what is it for?

File: 2.0/gradle/vividus-test-docker.gradle [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    from sourceSets.main.getResources().srcDirs
+}
+
+tasks.register('createTestsDockerfile', com.bmuschko.gradle.docker.tasks.image.Dockerfile) {
```
Comment (valfirst) [inline 2.0/gradle/vividus-test-docker.gradle:33]:
```suggestion
tasks.register('createDockerfile', com.bmuschko.gradle.docker.tasks.image.Dockerfile) {
```

File: 2.0/gradle/vividus-test-docker.gradle [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    def vividusLib = dependencies.find { matcher.call(it, 'vividus') }
+
+    def vividusVersion = vividusLib.version
+    if (vividusVersion == null) {
+        def vividusBomLib = dependencies.find { matcher.call(it, 'vividus-bom') }
+        vividusVersion = vividusBomLib.version
+    }
+
+    if (vividusVersion == null) {
+        throw new StopExecutionException('Unable to evaluate VIVIDUS version')
```
Comment (valfirst) [inline 2.0/gradle/vividus-test-docker.gradle:48]: need to make more descriptive

File: 2.0/gradle/vividus-test-docker.gradle [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+    dependsOn copyResources
+    from("vividus/vividus:${vividusVersion}")
+    copyFile('resources', '/vividus/resources/')
+}
+
+tasks.register('buildTestsImage', com.bmuschko.gradle.docker.tasks.image.DockerBuildImage) {
+    dependsOn createTestsDockerfile
+
+    def repositoryPath = System.getenv('DOCKER_REPOSITORY_PATH')
```
Comment (valfirst) [inline 2.0/gradle/vividus-test-docker.gradle:59]: why no to give the ability to specify the value in the project gradle script?

File: 2.0/gradle/vividus-test-docker.gradle [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+    dependsOn buildTestsImage
+    images = buildTestsImage.images
+
+    def registryUsername = System.getenv('DOCKER_REGISTRY_USERNAME')
+    def registryPassword = System.getenv('DOCKER_REGISTRY_PASSWORD')
+
+    if (registryUsername != null && registryPassword == null
+            || registryUsername == null && registryPassword != null) {
+        throw new StopExecutionException('The DDOCKER_REGISTRY_USERNAME and DOCKER_REGISTRY_PASSWORD env '
+            + 'variables should be either both set or unset')
```
Comment (valfirst) [inline 2.0/gradle/vividus-test-docker.gradle:77]: need to make more clear, like `Environment variable DOCKER_REGISTRY_USERNAME is set, but 
DOCKER_REGISTRY_PASSWORD is not` and vice versa

File: 2.0/gradle/vividus-test-docker.gradle (hunk 1 above)
Comment (uarlouski) [reply 2.0/gradle/vividus-test-docker.gradle:29]: docker plugin operates in output/docker folder by default

File: 2.0/gradle/vividus-test-docker.gradle [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+        classpath(group: 'com.bmuschko', name: 'gradle-docker-plugin', version: '9.3.4')
+    }
+}
+
+apply plugin: com.bmuschko.gradle.docker.DockerRemoteApiPlugin
+
+tasks.register('copyResources', Copy) {
+    dependsOn = [build, jar]
+    into layout.buildDirectory.dir('docker/libs')
+    from layout.buildDirectory.dir('libs')
```
Comment (valfirst) [inline 2.0/gradle/vividus-test-docker.gradle:31]:
```suggestion
    from layout.buildDirectory.dir('libs')
    into layout.buildDirectory.dir('docker/libs')
```

File: 2.0/gradle/vividus-test-docker.gradle [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+tasks.register('createDockerfile', com.bmuschko.gradle.docker.tasks.image.Dockerfile) {
+    dependsOn = [copyResources]
+    doFirst {
+        from("vividus/vividus:${project.ext.vividusVersion}")
+        copyFile('libs', '/vividus/libs/')
+    }
+}
+
+tasks.register('buildTestsImage', com.bmuschko.gradle.docker.tasks.image.DockerBuildImage) {
+    dependsOn createDockerfile
```
Comment (valfirst) [inline 2.0/gradle/vividus-test-docker.gradle:43]: let's follow some common convention for `dependsOn` declaration

File: 2.0/gradle/vividus-test-docker.gradle [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+    into layout.buildDirectory.dir('docker/libs')
+    from layout.buildDirectory.dir('libs')
+}
+
+tasks.register('createDockerfile', com.bmuschko.gradle.docker.tasks.image.Dockerfile) {
+    dependsOn = [copyResources]
+    doFirst {
+        from("vividus/vividus:${project.ext.vividusVersion}")
+        copyFile('libs', '/vividus/libs/')
+    }
```
Comment (valfirst) [inline 2.0/gradle/vividus-test-docker.gradle:39]: do we need to define entry point here?

**Review by valfirst — CHANGES_REQUESTED** [review]

File: 2.0/gradle/vividus-test-docker.gradle (hunk 8 above)
Comment (uarlouski) [reply 2.0/gradle/vividus-test-docker.gradle:39]: image
its inherited from parent image

File: 2.0/gradle/vividus-test-docker.gradle [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+}
+
+tasks.register('buildTestsImage', com.bmuschko.gradle.docker.tasks.image.DockerBuildImage) {
+    dependsOn [createDockerfile]
+
+    def repositoryPath = getProperty('DOCKER_REPOSITORY_PATH', 'docker.repository-path')
+    if (repositoryPath == null) {
+        throw new StopExecutionException('Please set repository path using \'docker.repository-path\' gradle property'
+            + ' or \'DOCKER_REPOSITORY_PATH\' env variable')
```
Comment (valfirst) [inline 2.0/gradle/vividus-test-docker.gradle:48]:
```suggestion
            + ' or \'DOCKER_REPOSITORY_PATH\' environemnt variable')
```

File: 2.0/gradle/vividus-test-docker.gradle [hunk 10]
```diff
@@ -0,0 +1,81 @@
...
+        gradlePluginPortal()
+    }
+    dependencies {
+        classpath(group: 'com.bmuschko', name: 'gradle-docker-plugin', version: '9.3.4')
+    }
+}
+
+apply plugin: com.bmuschko.gradle.docker.DockerRemoteApiPlugin
+
+tasks.register('copyResources', Copy) {
```
Comment (valfirst) [inline 2.0/gradle/vividus-test-docker.gradle:28]:
```suggestion
tasks.register('copyProjectLibrary', Copy) {
```

**Review by valfirst — CHANGES_REQUESTED** [review]

**Review by valfirst — APPROVED** [review]

**Pull request merged**